File: frontend/app/models/contract.ts
import { Client } from "@/app/models/client";
import { Advisor } from "@/app/models/advisor";

export type Contract = {
    id: string;
    referenceNumber: string;
    institution: string;
    signedDate: string;
    maturityDate: string;
    validUntilDate: string;
    client: Client | null;
    advisors: Advisor[];
    manager: Advisor | null;
};

File: backend/backend/Domain/Contract.cs
// // Created by Kateřina Plívová on 28.05.2025.

using backend.Domain;

public class Contract
{
    public int Id { get; set; }
    public string ReferenceNumber { get; set; }
    public string Institution { get; set; }
    public DateTime SignedDate { get; set; }
    public DateTime MaturityDate { get; set; }
    public DateTime ValidUntilDate { get; set; }

    public int ClientId { get; set; }
    public Client Client { get; }
    
    public int? ManagerId { get; set; }
    public Advisor Manager { get; set; }
    
    public List<Advisor> Advisors { get; set; }
}